test/test_xla_sharding.py

```diff
@@ -102,6 +102,20 @@ def test_deep_copy(self):
         torch_xla._XLAC._get_xla_sharding_spec(xt),
         torch_xla._XLAC._get_xla_sharding_spec(xt2))
 
+  def test_mark_step_with_sharding(self):
+    xt = torch.ones(2, 2).to(xm.xla_device())
+    xs.mark_sharding(xt, self._get_mesh((1, self.n_devices)),
+                     (0, 1))
+    sharding_spec = torch_xla._XLAC._get_xla_sharding_spec(xt)
+    xm.mark_step()  # resets IR value
+    self.assertEqual(sharding_spec, torch_xla._XLAC._get_xla_sharding_spec(xt))
+
+    xt.add_(1)  # inplace update
+    xm.mark_step()  # resets IR value
```

Comment:
I don't think you need the `mark_step` here? After the `xt.add_(1)`, if you print the IR and HLO by 
```
print(torch_xla._XLAC._get_xla_tensors_text([xt]))
print(torch_xla._XLAC._get_xla_tensors_hlo([xt]))
```
you should see the sharding spec on the output?

Reply:
It works either way, before or after -- and checks for the sharding annotation.